.vsts.pipelines/phases/ci-osx.yml

```diff
@@ -0,0 +1,85 @@
+parameters:
+  matrix:
+    Production: {}
+  phase: osx
+  queueDemands: Agent.OS -equals Darwin
+  queueName: DotNetCore-Mac
+
+phases:
+- phase: ${{ parameters.phase }}
+  variables:
+    # Prefix to distinguish artifacts from different legs.
+    artifactName: ${{ format('$(type)_{0}', parameters.phase) }}
+    logsDirectory: $(Build.ArtifactStagingDirectory)/logs
+    SOURCE_BUILD_SKIP_SUBMODULE_CHECK: true
+    # Default type, can be overridden by matrix legs.
+    type: Production
+  queue:
+    name: ${{ parameters.queueName }}
+    demands: ${{ parameters.queueDemands }}
+    timeoutInMinutes: 240
+    matrix:
```

Comment:
I don't understand this syntax.  Why can't this just be: matrix: ${{ parameters.matrix }}  ?

Reply:
I'll give it a try (I agree it'd make sense if that worked), I think I remember getting a validation error trying to add it like that. The insert syntax is documented here: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=vsts#insertion

Reply:
Nevermind, that works fine!